pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+    Linear = 0
+
+    Logarithmic = 1
+
+
+class SensorType(Enum):
+    """Enumeration to represent the selected sensor type for the power meter"""
+
+    #: Default (100% for all frequencies)
+    Default = 0
+
+    HP_8481A = 1
+
+    #: HP 8482A, 8482B, 8482H
+    HP_8482X = 2
+
+    HP_8483A = 3
+
+    HP_8481D = 4
+
+    HP_8485A = 5
+
+    HP_R8486A = 6
+
+    HP_Q8486A = 7
+
+    HP_R8486D = 8
+
+    HP_8487A = 9
+
+
+class OperatingMode(Enum):
+    """Enumeration to represent the operating mode the power meter is currently in"""
+
+    Normal = 0
+
+    Zeroing = 6
+
+    Calibration = 8
+
+
+class TriggerMode(Enum):
+    """Enumeration to represent the trigger mode the power meter is currently in"""
+
+    Hold = 0
+
+    FreeRunning = 3
+
+
+class GroupTriggerMode(Enum):
+    """Enumeration to represent the group execute trigger mode the power meter is currently in"""
+
+    Ignore = 0
+
+    TriggerImmediate = 1
+
+    TriggerDelay = 2
+
+
+class EventStatusRegister(IntFlag):
+    """Enumeration to represent the Event Status Register."""
+
+    #: The bit is set when the power meter's LINE switch is set from STDBY to ON
+    PowerOn = 128
+
+    #: This bit is set when an incorrect HP-IB code is sent to the power meter. For example,
+    # the command “QX” is a command error.
+    CommandError = 32
+
+    #: This bit is set when incorrect data is sent to the power meter. For example, the command
+    # “FR-3GZ” is an execution error.
+    ExecutionError = 16
+
+    #: This bit is set true whenever a measurement error (error 1-49) occurs.
+    DeviceDependentError = 8
+
+
+Errors = {
+    1: "Power meter cannot zero the sensor",
+    5: "Power meter cannot calibrate sensor",
+    11: "Input overload on sensor",
+    15: "Sensor’s zero reference has drifted negative",
+    17: "Input power on sensor is too high for current range",
+    21: "Power reading over high limit",
+    23: "Power reading under low limit",
+    31: "No sensor connected to the input",
+    33: "Both front and rear sensor inputs, have sensors connected (Option 002 or Option 003 only)",
+    50: "Entered cal factor is out of range",
+    51: "Entered offset is out of range",
+    52: "Entered range number is out of range",
+    54: "Entered recall register number is out of range",
+    55: "Entered storage register number is out of range",
+    56: "Entered reference cal factor is out of range",
+    57: "RAM ID check failure",
+    61: "Stack RAM failure",
+    62: "ROM checksum failure",
+    64: "RAM failure",
+    65: "Analog I/O PIA Failure",
+    66: "Keyboard and Display PIA Failure",
+    67: "Analog-to-Digital converter Failure",
+    68: "HP-IB failure",
+    69: "Timer failure",
+    70: "Keyboard/Display controller failure",
+    71: "Keyboard data failure",
+    72: "Data line to A3U26 is open.",
+    73: "Keyboard/Display controller self-test failure",
+    74: "Display not responding",
+    75: "Digital failure",
+}
+
+
+class StatusMessage:
```

Comment:
you meant to make it an Enum, didn't you?

Reply:
Not this one. That one is only for internal use.